Userland/Applications/Browser/History/HistoryModel.cpp

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2022, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "HistoryModel.h"
+#include <AK/FuzzyMatch.h>
+
+namespace Browser {
+
+void HistoryModel::set_items(AK::Vector<History::URLTitlePair> items)
+{
+    begin_insert_rows({}, m_entries.size(), m_entries.size());
+    m_entries = move(items);
+    end_insert_rows();
+
+    did_update(DontInvalidateIndices);
+}
+
+void HistoryModel::clear_items()
+{
+    begin_insert_rows({}, m_entries.size(), m_entries.size());
+    m_entries.clear();
+    end_insert_rows();
+
+    did_update(DontInvalidateIndices);
+}
+
+int HistoryModel::row_count(GUI::ModelIndex const& index) const
+{
+    if (!index.is_valid())
+        return m_entries.size();
+    return 0;
+}
+
+DeprecatedString HistoryModel::column_name(int column) const
+{
+    switch (column) {
+    case Column::Title:
+        return "Title";
+    case Column::URL:
+        return "URL";
+    case Column::__Count:
+        return {};
```

Comment:
Bit odd to handle the `__Count` case. Maybe just put `default: VERIFY_NOT_REACHED();` instead?

Reply:
Then it's odd there too :^)

Reply:
Fixed